rest_api/test/samples/pipeline/test_pipeline.yaml

```diff
@@ -0,0 +1,103 @@
+version: '0.7'
```

Comment:
Are these yamls + `rest/api/test/test_rest_api.py` commited by purpose to this PR?

Reply:
Yes. In fact along with this PR I also included this changes: #1705. However, removing the `rest_api` and `ui` from the package (they were polluting the namespace, so it's a good change IMHO) they cannot be imported from the `test` folder easily. So I decided to see what happens if I move the relative tests in their own directories, so `test_res_api.py` goes in `rest_api/test/` and `test_ui_utils.py` goes in `ui/test/`. This way I have no problem with the imports.
What do you think?

Reply:
Ah ok, I see. Removing them from the package is great. 
I see two (small) downsides of the solution for the tests: 
- A bit counter-intuitive structure as all our other tests are central (and not in the different modules). New contributors might be confused about where to add a test for API or UI
- How would we run all tests locally? Would something like `pytest . ` from the repository root collect all tests? Haven't tested it but I guess it will create trouble with our `pytest.ini` and maybe the custom `--document_store_type` argument.

Reply:
`pytest --co` (collect-only option) finds the tests successfully, so I don't think it will be a problem. Running `pytest` in the root directory will run all tests, as it was before. I agree it might not be immediate where the rest api and ui tests are, but on the other hand a person that has modified the `rest_api/` or `ui/` folders content should notice the `test/` folder right there I hope :grin: Overall I don't think it will cause trouble. Ironically it was harder to convince the CI to find them, but it should be a solved issue now.